pyproject.toml

```diff
@@ -43,6 +43,20 @@ email = 'skimage-core@discuss.scientific-python.org'
 file = 'LICENSE.txt'
 
 [project.optional-dependencies]
+data = ['pooch>=1.6.0']
+optional = [
+    'SimpleITK; sys_platform != "emscripten"',  # not packaged for Pyodide yet
+    'astropy>=6.0',
+    'cloudpickle>=3.0',  # necessary to provide the 'processes' scheduler for dask
+    'dask[array]>=2023.2.0',
+    'matplotlib>=3.7',
+    'pooch>=1.6.0; sys_platform != "emscripten"',  # requires stack switching with alternative pytest runner
+    'pyamg>=5.2; sys_platform != "emscripten"',  # not packaged for Pyodide yet
+    'PyWavelets>=1.6',
+    'scikit-learn>=1.2',
+]
```

Comment:
Not sure whether you looked into this already. Is there some way to define "optional" as a dependency group as well and reference it here? Then everything would be in one place. :thinking: 

I skimmed through the [pyproject.toml spec](https://packaging.python.org/en/latest/specifications/pyproject-toml/#dependencies-optional-dependencies) but didn't find anything useful.

Reply:
Fine with me.